src/Neo/SmartContract/Native/ContractEventAttribute.cs

```diff
@@ -17,35 +17,48 @@ namespace Neo.SmartContract.Native
 {
     [DebuggerDisplay("{Descriptor.Name}")]
     [AttributeUsage(AttributeTargets.Constructor, AllowMultiple = true)]
-    internal class ContractEventAttribute : Attribute
+    internal class ContractEventAttribute : Attribute, IHardforkActivable
     {
         public int Order { get; init; }
         public ContractEventDescriptor Descriptor { get; set; }
         public Hardfork? ActiveIn { get; init; } = null;
+        public Hardfork? DeprecatedIn { get; init; } = null;
 
         public ContractEventAttribute(Hardfork activeIn, int order, string name,
             string arg1Name, ContractParameterType arg1Value) : this(order, name, arg1Name, arg1Value)
         {
             ActiveIn = activeIn;
         }
 
+        public ContractEventAttribute(Hardfork activeIn, int order, string name,
+            string arg1Name, ContractParameterType arg1Value, Hardfork deprecatedIn) : this(activeIn, order, name, arg1Name, arg1Value)
+        {
+            DeprecatedIn = deprecatedIn;
+        }
+
         public ContractEventAttribute(int order, string name, string arg1Name, ContractParameterType arg1Value)
         {
             Order = order;
             Descriptor = new ContractEventDescriptor()
             {
                 Name = name,
-                Parameters = new ContractParameterDefinition[]
-                {
+                Parameters =
+                [
                     new ContractParameterDefinition()
                     {
                         Name = arg1Name,
                         Type = arg1Value
                     }
-                }
+                ]
             };
         }
 
+        public ContractEventAttribute(int order, string name, string arg1Name, ContractParameterType arg1Value, Hardfork deprecatedIn)
+            : this(order, name, arg1Name, arg1Value)
+        {
+            DeprecatedIn = deprecatedIn;
+        }
```

Comment:
Could you make one more constructor for notifications with two args?

Reply:
Done